docs/plugins.md

```diff
@@ -14,183 +14,171 @@ Looking for help using Snowpack in your project?
 
-#### Example
+| Key                |    Type    | Description                                                                               |
+|:-------------------|:----------:|:------------------------------------------------------------------------------------------|
+| `name`             |  `string`  | The plugin name. This will be shown whenever there are errors or messages.                |
+| `input`            | `string[]` | An array of file extensions this plugin listens for (e.g. `['.js', '.jsx', '.ts']`)       |
+| `output`           | `string[]` | An array of file extensions this plugin may output (e.g. `['.js', '.css']`)               |
+| `knownEntrypoints` | `string[]` | Array of npm dependencies this plugin needs to include all of (e.g. `['svelte/internal']` |
+| `build()`          | `function` | An async function that takes file info and returns transformed code.                      |
 
-```ts
-knownEntrypoints: ["svelte/internal"], // ensures svelte/internal exists in web_modules at the end
-```
+Let’s look at some examples to see the basics in action:
+
+#### Single file transformation
 
-### defaultBuildScript
+For our first example, we’ll look at transforming files 1:1.
 
-```ts
-defaultBuildScript?: string
+```js
+module.exports = (snowpackConfig, pluginOptions) => ({
+  name: 'my-commenter-plugin',
+  input: ['.js', '.mjs', '.jsx', '.ts', '.tsx'],
+  async build({ contents, extension }) {
+    return `/* I’m a comment! */ ${contents}`;
+  }
+})
 ```
 
-The build script that will be used for this plugin, if none is provided by the user. If your plugin uses the `build()` method, this is what controls which files your build command will match against by default.
+This simple plugin takes all JavaScript & TypeScript files (`['.js', '.mjs', '.jsx', '.ts', '.tsx']`), and prepends a simple comment (`/* I’m a comment */`) to the beginning of each file. Even though this is a contrived example, it introduces us to the plugin API.
 
-If you don't include this, the user will need to define the build script themselves for the plugin to do anything.
+Our plugin **name** should be the same as name as the package (or if it’s a local plugin, give it a name).
 
-#### Example
+The **input** of the plugin is an array of file extensions we want to listen to. Notice we could have only specified `['.js']` if we wanted to, but in our example we wanted to listen for JSX and TypeScript files as well.
 
-```ts
-defaultBuildScript: "build:vue", // hooks into the user’s build:vue script automatically unless they manually override this
-```
 
-### build()
-
-```ts
-build?: async ({
-  filePath: string,
-  contents: string,
-  isDev: boolean,
-}) => null | {
-  result: string;
-  resources?: {css?: string};
-};
-```
+And lastly, the meat of our plugin lives in `build()`. We took the `contents` of the original file, and added our own comment to the beginning. And that’s it! That string turns back into JavaScript. ✨ Magic! 🦄
```

Comment:
super nit:

```suggestion
And lastly, the meat of our plugin lives in `build()`. We took the `contents` of the original file and added our comment to the beginning. And that’s it! That string turns back into JavaScript. ✨ Magic! 🦄
```

Reply:
I am really digging your copy edits! I agree with all of them.

![](https://media.giphy.com/media/xP9tNSAQEZSJG/giphy.gif)

Reply:
https://media1.giphy.com/media/qr0FOwPgUk3Cw/giphy.gif